Core/FitResult.cpp

```diff
@@ -4,232 +4,148 @@
 // https://github.com/ComPWA/ComPWA/license.txt for details.
 
 #include "Core/FitResult.hpp"
+
 #include "Core/Logging.hpp"
+#include "Core/TableFormatter.hpp"
+#include "Core/Utils.hpp"
 
-namespace ComPWA {
+#include <boost/serialization/nvp.hpp>
+#include <boost/serialization/vector.hpp>
 
-void FitResult::writeText(std::string filename) {
-  std::ofstream myfile;
-  myfile.open(filename, std::ios::app);
-  genOutput(myfile);
-  myfile.close();
-  return;
-};
-
-void FitResult::writeSimpleText(std::string filename) {
-  std::ofstream myfile;
-  myfile.open(filename);
-  genSimpleOutput(myfile);
-  myfile.close();
-  return;
-};
-
-double FitResult::shiftAngle(double v) {
-  double originalVal = v;
-  double val = originalVal;
-  while (val > M_PI)
-    val -= 2 * M_PI;
-  while (val < (-1) * M_PI)
-    val += 2 * M_PI;
-  if (val != originalVal)
-    LOG(INFO) << "shiftAngle(): shifting parameter from " << originalVal
-              << " to " << val << "!";
-  return val;
-}
-
-void FitResult::genSimpleOutput(std::ostream &out) {
-  for (auto p : FinalParameters.doubleParameters()) {
-    out << p->value() << " ";
-    if (p->hasError())
-      out << p->avgError() << " ";
-  }
-  out << "\n";
+#include <iomanip>
 
-  return;
-}
-
-void FitResult::setFinalParameters(ParameterList &finPars) {
-  FinalParameters.DeepCopy(finPars);
-}
+namespace ComPWA {
 
-void FitResult::setTrueParameters(ParameterList &truePars) {
-  TrueParameters.DeepCopy(truePars);
+std::string makeFitParameterString(ComPWA::FitParameter<double> p) {
+  std::stringstream ss;
+  ss << std::setprecision(10);
```

Comment:
To cout full precision of a variable, we can do it like this:
https://stackoverflow.com/questions/554063/how-do-i-print-a-double-value-with-full-precision-using-cout
double d = 100.0000000000005;
int precision = std::numeric_limits<double>::max_digits10;
std::cout << std::setprecision(precision) << d << std::endl;
I think this will be better than a fixed precision(but maybe the printed number is very long...)

Reply:
Do we need/want full precision there?